ngraph/test/backend/bucketize.in.cpp

```diff
@@ -0,0 +1,71 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "gtest/gtest.h"
+#include "ngraph/ngraph.hpp"
+#include "util/engine/test_engines.hpp"
+#include "util/test_case.hpp"
+#include "util/test_control.hpp"
+
+NGRAPH_SUPPRESS_DEPRECATED_START
+
+using namespace std;
+using namespace ngraph;
+
+static string s_manifest = "${MANIFEST}";
+using TestEngine = test::ENGINE_CLASS_NAME(${BACKEND_NAME});
+
+NGRAPH_TEST(${BACKEND_NAME}, bucketize_right_edge)
+{
+    Shape data_shape{10, 1};
+    Shape bucket_shape{4};
+
+    const auto data = make_shared<op::Parameter>(element::f32, data_shape);
+    const auto buckets = make_shared<op::Parameter>(element::i32, bucket_shape);
+    const auto bucketize = make_shared<op::v3::Bucketize>(data, buckets, element::i32, true);
+    const auto f = make_shared<Function>(bucketize, ParameterVector{data, buckets});
+
+    vector<float> data_vect = {8.f, 1.f, 2.f, 1.1f, 8.f, 10.f, 1.f, 10.2f, 0.f, 20.f};
```

Comment:
Can we test value higher than bucket max (20) in the left and right edge test case?

Reply:
bucketize is an element-wise operator. For each element of the input data will check upper bound if right edge or lower bound if left edge. Adding another element to data_vect, for instance 21 will result in output element 4 for right edge. This can be seen in bucketize_left_edge test case, where 20 produces index 4 as it is left edge. As a summary adding another element greater than bucket max does not add any value to the unit test.